`or` block in `for` declaration does not work

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
**OS:** 

```
OS: linux, "Arch Linux"
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700F CPU @ 3.00GHz
CC version: cc (GCC) 10.2.0

getwd: /storage/home/zero/Documents/programming/vemulator/assembler
vmodules: /home/zero/.vmodules
vroot: /usr/lib/vlang
vexe: /usr/lib/vlang/v
vexe mtime: 2021-04-13 21:20:37
is vroot writable: false
is vmodules writable: true
V full version: V 0.2.2 a2a18ef.8c95f07

Git version: git version 2.31.1
Git vroot status: weekly.2021.14-79-g8c95f075
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81-dirty
```

**What did you do?**

```v
import os 
for d in os.read_lines("main.v") or { panic("not found") } { 
    println(d) 
} 
```
**What did you expect to see?**

Either a `panic`, or the compiler telling me "hey dummy, you can't `or` right there"
 
**What did you see instead?**

```
==================
/tmp/v/.noprefix.01F36M1C1KGYS3TAXPYMKDRH5P.vrepl_temp.7985419067470181272.tmp.c:14773: error: ';' expected (got "}")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
>>> 
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5906,6 +5906,9 @@ fn (mut g Gen) or_block(var_name string, or_block ast.OrExpr, return_type ast.Ty
 			g.indent--
 		} else {
 			g.stmts(stmts)
+			if stmts.len > 0 && stmts[or_block.stmts.len - 1] is ast.ExprStmt {
+				g.writeln(';')
+			}
 		}
 	} else if or_block.kind == .propagate {
 		if g.file.mod.name == 'main' && (isnil(g.fn_decl) || g.fn_decl.is_main) {
@@ -5936,7 +5939,7 @@ fn (mut g Gen) or_block(var_name string, or_block ast.OrExpr, return_type ast.Ty
 			}
 		}
 	}
-	g.write('}')
+	g.writeln('}')
 }
 
 // `a in [1,2,3]` => `a == 1 || a == 2 || a == 3`

--- a/vlib/v/tests/for_in_optional_test.v
+++ b/vlib/v/tests/for_in_optional_test.v
@@ -1,0 +1,8 @@
+import os
+
+fn test_for_in_optional() {
+	for d in os.read_lines(@FILE) or { panic('not found') } {
+		println(d)
+	}
+	assert true
+}

